src/NuGet.Core/NuGet.Configuration/Settings/Settings.cs

```diff
@@ -41,11 +41,11 @@ public class Settings : ISettings
             new[] { "*.config" } :
             new[] { "*.Config", "*.config" };
 
-        private readonly Dictionary<string, VirtualSettingSection> _computedSections;
+        public Dictionary<string, VirtualSettingSection> ComputedSections { get; }
```

Comment:
It looks like this class is designed to hide _computedSections from external callers and GetSection is supposed to be used to obtain data about the setting section. By exposing this ComputedSection publicly, we are changing the encapsulation of this class which we should be cautious about doing.

Reply:
Initially, I was using a method that returned the value of `_computedSections` to access this value, but [Andy](https://github.com/NuGet/NuGet.Client/pull/5126#discussion_r1167389921) and [Donnie](https://github.com/NuGet/NuGet.Client/pull/5126#discussion_r1167154519) suggested using an auto property to get the value instead. I wasn't able to figure out a way to use `GetSection`, since I don't know which section names might exist in the settings I'm using.

Reply:
I would vote to have a new method on `Settings` that returns an `IEnumerable<string>` which is just `_computedSections.Keys`